lib/grpc/client/resolver/dns.ex

```diff
@@ -0,0 +1,55 @@
+defmodule GRPC.Client.Resolver.DNS do
+  @moduledoc """
+  DNS Resolver for gRPC targets, supporting dynamic updates via a GenServer.
+
+  Resolves `dns://host[:port]` targets. Fetches A/AAAA records and optional
+  `_grpc_config.<host>` TXT records for ServiceConfig.
+
+  This implementation maintains an internal cache of addresses and service config,
+  and refreshes them periodically.
+  """
+  @behaviour GRPC.Client.Resolver
+
+  alias GRPC.Client.ServiceConfig
+
+  @impl GRPC.Client.Resolver
+  def resolve(target) do
+    uri = URI.parse(target)
+    host = uri.host || target
+    port = uri.port || 50051
+    host_charlist = String.to_charlist(host)
```

Comment:
Shouldn't the lookup accept strings and specific adapters convert to charlists as needed instead? Charlists are not very elixir-y IMO

Reply:
Done!